Realm/RLMSyncUser.mm

```diff
@@ -529,6 +526,11 @@ + (void)_performLoginForDirectAccessTokenCredentials:(RLMSyncCredentials *)crede
     NSString *identity = credentials.userInfo[kRLMSyncIdentityKey];
     std::shared_ptr<SyncUser> sync_user;
     if (serverURL) {
+        NSString *scheme = serverURL.scheme;
+        if (![scheme isEqualToString:@"http"] && ![scheme isEqualToString:@"https"]) {
+            @throw RLMException(@"The Realm Object Server authentication URL for this user, \"%@\", is improperly"
+                                @" configured. It must begin with http:// or https://.", serverURL);
```

Comment:
I feel like "invalid" makes more sense in this error than "improperly configured".